app/src/main/java/com/android/voyageur/ui/trip/activities/Activities.kt

```diff
@@ -94,23 +135,36 @@ fun ActivitiesScreen(
             selectedItem = navigationActions.currentRoute(),
             userViewModel = userViewModel)
       },
-      topBar = { // TODO: include the Search in TopAppBar
+      topBar = {
         TopAppBar(
             title = {
-              Text("Activities bar")
-            }, // This is a hardcoded string, as it should be replaced by the search bar
+              Box(
+                  modifier = Modifier.fillMaxWidth().fillMaxHeight(),
+                  contentAlignment = Alignment.CenterStart) {
+                    TextField(
+                        value = searchQuery,
+                        onValueChange = { searchQuery = it },
+                        placeholder = { Text("Search activities...") },
+                        modifier = Modifier.fillMaxWidth().testTag("searchField"),
+                        singleLine = true,
+                        shape = RoundedCornerShape(10.dp),
+                        keyboardOptions = KeyboardOptions(imeAction = ImeAction.Search),
+                        keyboardActions = KeyboardActions(onSearch = {}))
+                  }
+            },
```

Comment:
It would be nice to have this search bar on the By day activity screen as well, for consistency, wdyt? This screen is the one that pops up when you click on a certain day in the schedule view. Same for the filter button. You could make these a composable and move it to ActivityUtils maybe, import it from there to the ByDay screen. This can also be a separate PR if you prefer, since it touches the filter button as well.